hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/OutboxInternal.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.execution;
+
+import com.hazelcast.jet.core.Outbox;
+
+public interface OutboxInternal extends Outbox {
+
+    /**
+     * Resets the outbox so that it is available to receive another batch of
+     * items after any {@code offer()} method previously returned {@code
+     * false}.
```

Comment:
We use it to limit the number of items emitted in one run in case the downstream manages to drain the queue quickly enough, to avoid accepting more items forever.

Reply:
I just literally don't understand it. I would have to look at the implementation to understand what this method does. 

Is it true that you can add at most N items to the outbox after a `reset()` call, regardless of how much space it has left?

Reply:
Yes, it is true. The `batchSize` is a constructor parameter. Seems that the comment gave you a correct clue, it was just surprising :). 
Also whenever `offer` returns false, you must not try to offer again, including the same item, until `reset()` is called. And `reset()` is a private api, ProcessorTasklet calls it.